sdk/eventhub/event-hubs/test/utils/setup.ts

```diff
@@ -10,25 +10,28 @@ import { resolve as resolvePath } from "path";
 import type { GlobalSetupContext } from "vitest/node";
 import { EnvVarKeys } from "./constants.js";
 import * as MOCKS from "./constants.js";
+import type { AzureLogLevel } from "@azure/logger";
 
 declare module "vitest" {
   type MyEnvVarKeys = {
-    [K in keyof typeof EnvVarKeys]: string;
-  };
+    [K in keyof Omit<typeof EnvVarKeys, "AZURE_LOG_LEVEL">]: string;
+  } & { AZURE_LOG_LEVEL: AzureLogLevel | undefined };
   export interface ProvidedContext extends MyEnvVarKeys {}
 }
 
-const defaultLogLevel = "error";
-
-function assertEnvironmentVariable(key: string): string {
+function assertEnvironmentVariable<K extends EnvVarKeys.AZURE_LOG_LEVEL>(
+  key: K,
+): AzureLogLevel | undefined;
+function assertEnvironmentVariable(key: string): string;
+function assertEnvironmentVariable(key: string): string | undefined {
```

Comment:
nitpick: should the implementation signature include all overloads' return types? i.e., `AzureLogLevel | string | undefined;` even though `string | undefined` is equivalent.

Reply:
It bothers me a little for an `assertXxxx` function to allow missing value but I guess it's fine here for testing.